hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -505,7 +514,9 @@ public class HoodieWriteConfig extends HoodieConfig {
   private HoodieMetadataConfig metadataConfig;
   private HoodieMetastoreConfig metastoreConfig;
   private HoodieCommonConfig commonConfig;
+  private HoodieStorageConfig storageConfig;
   private EngineType engineType;
+  private HoodieRecordMerger recordMerger;
```

Comment:
I don't think we need to hold `recordMerger` -- we should instantiate it on the fly

Reply:
getRecordMerger will be called more than once for getting recordType(SPARK, AVRO). Holding recordMerger will be better? Or we can make it lazy loading.

Reply:
Let's avoid premature optimizations -- `RecordMerger` is specifically designed as stateless component and as such is lightweight enough to be initialized on-demand. Only thing we need to take care of is just making sure we don't init it for every record.

The problem with caching it inside `HoodieWriteConfig` is that HoodieWriteConfig has nothing to do w/ RecordMerger and we should not open this can of warm when it's used as a state carriage for other components.